### PR TITLE
MBS-9086 Fix broken screenshots tests recording

### PR DESCRIPTION
**Pull request by eugene-krivobokov**
blocker for #632

### CODE REVIEW

File: subprojects/gradle/design-screenshots/src/main/kotlin/com/avito/plugin/DeviceSearch.kt [hunk 1]
```diff
@@ -1,35 +0,0 @@
...
-    ) : DeviceSearch {
-        override fun getDevice(): Either<Exception, AdbDevice> {
-            adbDevicesManager.connectedDevices().let { set ->
-                val either = when (set.size) {
-                    0 -> {
-                        val exception = Exception("There are no connected devices")
-                        Either.left(exception)
-                    }
-                    1 -> {
-                        Either.right(set.first() as AdbDevice)
```
Comment (eugene-krivobokov) [inline subprojects/gradle/design-screenshots/src/main/kotlin/com/avito/plugin/DeviceSearch.kt:23]: Here was the unsafe cast:
```
Execution failed for task ':design-app:recordScreenshots'.
> class com.avito.runner.service.worker.device.adb.AdbDeviceParams cannot be cast to class com.avito.runner.service.worker.device.adb.AdbDevice (com.avito.runner.service.worker.device.adb.AdbDeviceParams and com.avito.runner.service.worker.device.adb.AdbDevice are in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @eca88ce)
```

File: subprojects/gradle/design-screenshots/src/main/kotlin/com/avito/plugin/DeviceProvider.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    private val adbDevicesManager: DevicesManager,
+    private val logger: CILogger
+) : DeviceProvider {
+
+    override fun getDevice(): AdbDevice {
+        val adbDeviceParams = connectedDeviceParams()
+
+        val serial: Serial.Local = when {
+            adbDeviceParams.id is Serial.Local -> adbDeviceParams.id as Serial.Local
+            adbDeviceParams.id is Serial.Remote ->  Serial.Local(adbDeviceParams.id.value) // TODO: support model of locally connected device
```
Comment (eugene-krivobokov) [inline subprojects/gradle/design-screenshots/src/main/kotlin/com/avito/plugin/DeviceProvider.kt:28]: Didn't change in this PR, but it seems a valid case.
It's helpful to connect emulator from local container. Especially for screenshots test.

**Review by dsvoronin — APPROVED** [review]

File: subprojects/gradle/design-screenshots/src/main/kotlin/com/avito/plugin/DeviceProvider.kt (hunk 2 above)
Comment (sergioboi) [reply subprojects/gradle/design-screenshots/src/main/kotlin/com/avito/plugin/DeviceProvider.kt:28]: As I know, screenshot generating works only at the `Local emulator`

**Comment by sergioboi** [comment]
infra version of screenshot tests is not used now. So we could not be blocked

File: subprojects/gradle/design-screenshots/src/main/kotlin/com/avito/plugin/DeviceProvider.kt (hunk 2 above)
Comment (eugene-krivobokov) [reply subprojects/gradle/design-screenshots/src/main/kotlin/com/avito/plugin/DeviceProvider.kt:28]: The term "local" is confusing. 
I mean local emulator inside Docker container. It is connected to adb as "localhost:5555".
This is the easiest way for me to use screenshots tests locally.

Comment (sergioboi) [reply subprojects/gradle/design-screenshots/src/main/kotlin/com/avito/plugin/DeviceProvider.kt:28]: As I know @DEADMC made screenshots by the emulator without docker on the local machine

**Pull request merged**